### ISSUE TITLE
Rule Change: (fill in)

### ISSUE THREAD

**Issue by is-harshul** [eslint/eslint#17543]
### What rule do you want to change?

`prefer-destructuring`

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
const {foo, boo} = barObj; // <- Error here
console.log(foo, boo);


// Enforcing this
console.log(barObj.foo, barObj.boo);
```

### What does the rule currently do for this code?

Enforces to destructure the object/array.

### What will the rule do after it's changed?

Will stop the developers from destructuring the object.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @is-harshul, thanks for the issue!

It wouldn't make sense for _prefer_-destructuring rule to disallow destructuring.

If you want to disallow object destructuring, you can use the [no-restricted-syntax](https://eslint.org/docs/latest/rules/no-restricted-syntax) rule with the following selector:

```
ObjectPattern
```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gICAgc2VsZWN0b3I6IFwiT2JqZWN0UGF0dGVyblwiLFxuICAgIG1lc3NhZ2U6IFwiT2JqZWN0IGRlc3RydWN0dXJpbmcgaXMgZGlzYWxsb3dlZC5cIlxufV0qL1xuXG5jb25zdCB7Zm9vLCBib299ID0gYmFyT2JqOyAvLyA8LSBFcnJvciBoZXJlXG5jb25zb2xlLmxvZyhmb28sIGJvbyk7XG5cblxuLy8gRW5mb3JjaW5nIHRoaXNcbmNvbnNvbGUubG9nKGJhck9iai5mb28sIGJhck9iai5ib28pOyIsIm9wdGlvbnMiOnsiZW52Ijp7ImVzNiI6dHJ1ZX0sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQifX19)

**Comment by is-harshul** [comment]
thanks man, will check this
